stdlib/public/core/Character.swift

```diff
@@ -93,7 +93,7 @@ public struct Character :
     _precondition(
       s._core.count != 0, "Can't form a Character from an empty String")
     _precondition(
-      s.startIndex.successor() == s.endIndex,
+      s.characters.next(s.startIndex) == s.endIndex,
```

Comment:
I guess we should expose next(Index), etc. on String to make the API symmetric with having startIndex and endIndex?